libs/api/domains/health-directorate/src/lib/health-directorate.service.ts

```diff
@@ -24,84 +20,78 @@ export class HealthDirectorateService {
   ) {}
 
   /* Organ Donation */
-  async getDonorStatus(auth: Auth): Promise<DonorStatus> {
+  async getDonorStatus(auth: Auth, locale: Locale): Promise<Donor> {
+    const lang: organLocale = locale === 'is' ? organLocale.Is : organLocale.En
     const data: OrganDonorDto | null =
-      await this.organDonationApi.getOrganDonation(auth)
+      await this.organDonationApi.getOrganDonation(auth, lang)
+    // Fetch organ list to get all names in correct language to sort out the names of the organs the user has limitations for
```

Comment:
No possibility of the organ donation returning null? 
Just wondering if the organ donation returns null, and we return an empty donor object here. Won't it be a bit awkward to null check in the front end?

Reply:
We always check if the data is there or not in the frontend, I dont see the difference?

Reply:
What i mean is the the organ donation screens null checks like this
```
 {!error && !loading && data === null && (
        <Problem type="no_data" noBorder={false} />
      )}
```
But it seems like the `getDonorStatus` query would return an empty `Donor` if there isn't any data instead of a null, this the check above wouldn't ever trigger.

Reply:
Aaa I see, I'll add the null return value 🫡